test_parse_strict_noncritical_error_default fails on 3/4 exotic archs with AssertionError: Regex pattern did not match

Example log: https://github.com/astropy/astropy/actions/runs/11240991355/job/31251556253?pr=17139

I think this is caused by one of the OGIP changes that @eerovaher made recently (and @mhvk probably reviewed it). Somehow that change broke a test but only in the matrix of exotic archs job that uses some system libraries (but at a glance I don't see how sys lib can cause this error). This should be resolved before v7.0 is released.

https://github.com/astropy/astropy/blob/eb9894925bd1b0f2037f18a4b0ab0a08bcc849e9/.github/workflows/ci_cron_weekly.yml#L166

```
      def test_parse_strict_noncritical_error_default():
          with pytest.raises(
              ValueError,
              match=(
                  r"^if 'm\(s\)' was meant to be a multiplication, it should have been "
                  r"written as 'm \(s\)'\.\n"
                  "If you cannot change the unit string then try specifying the "
                  r"'parse_strict' argument\.$"
              ),
          ):
  >           assert u.Unit("m(s)", format="ogip")
  
  E       AssertionError: Regex pattern did not match.
  E        Regex: "^if 'm\\(s\\)' was meant to be a multiplication, it should have been written as 'm \\(s\\)'\\.\\nIf you cannot change the unit string then try specifying the 'parse_strict' argument\\.$"
  E        Input: "if 'm(s)' was meant to be a multiplication, it should have been written as 'm (s)'."
  
  astropy/units/tests/test_units.py:263: AssertionError
```

```
  _________________ test_parse_strict_noncritical_error_default __________________
  
  self = <class 'astropy.units.core.Unit'>, s = 'm(s)', represents = None
  format = 'ogip', namespace = None, doc = None, parse_strict = 'raise'
```

--- a/astropy/units/tests/test_units.py
+++ b/astropy/units/tests/test_units.py
@@ -12,6 +12,7 @@ from numpy.testing import assert_allclose
 
 from astropy import constants as c
 from astropy import units as u
+from astropy.tests.helper import PYTEST_LT_8_0
 from astropy.units import utils
 from astropy.utils.compat.optional_deps import HAS_DASK
 from astropy.utils.exceptions import AstropyDeprecationWarning
@@ -260,15 +261,20 @@ def test_parse_strict_noncritical_error(parse_strict, expectation):
 
 
 def test_parse_strict_noncritical_error_default():
-    with pytest.raises(
-        ValueError,
-        match=(
+    if PYTEST_LT_8_0:
+        # pytest < 8 does not know how to deal with `Exception.add_note`
+        match = (
+            r"^if 'm\(s\)' was meant to be a multiplication, it should have been "
+            r"written as 'm \(s\)'\.$"
+        )
+    else:
+        match = (
             r"^if 'm\(s\)' was meant to be a multiplication, it should have been "
             r"written as 'm \(s\)'\.\n"
             "If you cannot change the unit string then try specifying the "
             r"'parse_strict' argument\.$"
-        ),
-    ):
+        )
+    with pytest.raises(ValueError, match=match):
         assert u.Unit("m(s)", format="ogip")
 
 

